packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -581,34 +587,41 @@ export function SelectPanel({
           />
           {footer ? (
             <div className={classes.Footer}>{footer}</div>
-          ) : showCancelSaveButtons ? (
+          ) : secondaryAction !== undefined || showCancelSaveButtons ? (
             /* Save and Cancel buttons are only useful for multiple selection, single selection instantly closes the panel */
-            <div className={clsx(classes.Footer, classes.ResponsiveFooter)}>
-              {onCancel && (
-                <Button
-                  size="medium"
-                  onClick={() => {
-                    onCancel()
-                    onCancelRequested()
-                  }}
-                >
-                  Cancel
-                </Button>
-              )}
-              <Button
-                block={onCancel === undefined}
-                variant="primary"
-                size="medium"
-                onClick={() => {
-                  if (isSingleSelectModal) {
-                    const singleSelectOnChange = onSelectedChange as SelectPanelSingleSelection['onSelectedChange']
-                    singleSelectOnChange(intermediateSelected)
-                  }
-                  onClose(variant === 'modal' ? 'selection' : 'click-outside')
-                }}
-              >
-                Save
-              </Button>
+            <div data-footer-layout={footerLayout} className={clsx(classes.Footer, classes.ResponsiveFooter)}>
+              <div data-footer-layout={footerLayout} className={classes.SecondaryAction}>
+                {secondaryAction}
+              </div>
+              {showCancelSaveButtons ? (
+                <div className={classes.CancelSaveButtons}>
+                  {onCancel ? (
+                    <Button
+                      size="medium"
+                      onClick={() => {
+                        onCancel()
+                        onCancelRequested()
+                      }}
+                    >
+                      Cancel
+                    </Button>
+                  ) : null}
+                  <Button
+                    block={onCancel === undefined}
+                    variant="primary"
+                    size="medium"
+                    onClick={() => {
+                      if (isSingleSelectModal) {
+                        const singleSelectOnChange = onSelectedChange as SelectPanelSingleSelection['onSelectedChange']
+                        singleSelectOnChange(intermediateSelected)
+                      }
+                      onClose(variant === 'modal' ? 'selection' : 'click-outside')
+                    }}
+                  >
+                    Apply
```

Comment:
```suggestion
                    Save
```

Reply:
@emilybrick which label should we go for? Figma says apply

Reply:
Sorry I missed this on Fri - it should say Save